gen/gen.go

```diff
@@ -18,41 +18,40 @@ func main() {
 	ts.Accumulate(schema.SpawnList("ListString", "String", false))
 	ts.Accumulate(schema.SpawnLink("Link"))
 	ts.Accumulate(schema.SpawnStruct("PersonInfo", []schema.StructField{
-		schema.SpawnStructField("Name", "String", false, false),
-		schema.SpawnStructField("Email", "String", false, false),
-		schema.SpawnStructField("Date", "String", false, false),
-		schema.SpawnStructField("Timezone", "String", false, false),
-	}, schema.SpawnStructRepresentationStringjoin(" ")))
+		schema.SpawnStructField("date", "String", false, false),
+		schema.SpawnStructField("timezone", "String", false, false),
+		schema.SpawnStructField("email", "String", false, false),
+		schema.SpawnStructField("name", "String", false, false),
+	}, schema.SpawnStructRepresentationMap(map[string]string{})))
```

Comment:
why did you change this to a map rather than a stringjoin?

Reply:
Because it's not being used as a stringjoin in any way? We have a custom `PersonInfo#GitString()` that's used to encode it in the git format and `parsePersonInfo()` to parse the input string for decoding. Stringjoin here is only going to mean that when it's encoded with dag-cbor or dag-json that it comes out as `<date> <timezone> <email> <name>`, which IMO is lossy and probably not what we're wanting out of this. Unless I'm missing something?

Reply:
`<date> <timezone> <email> <name>` is the git format though. I would hope that wouldn't be lossy.

Reply:
it's `$name <$email> $maybedate $maybetimezone` - it's lossy because the email has a `<>` around it and we can use the `<` to delineate the name which itself can contain any number of spaces, so simple stringjoin isn't enough

Reply:
Agree this seems better to do at the codec layer and yield up to data model as a map.

If I was going to transliterate some git data to dag-json, I'd probably want this to become a map in dag-json.